pkg/api/dashboard_test.go

```diff
@@ -95,6 +98,14 @@ func newTestLive(t *testing.T) *live.GrafanaLive {
 func TestDashboardAPIEndpoint(t *testing.T) {
 	t.Run("Given a dashboard with a parent folder which does not have an ACL", func(t *testing.T) {
 		setUp := func() *testState {
+			libraryelements.NewService = func(store *sqlstore.SQLStore) libraryelements.LibraryElementService {
+				return &mockLibraryElementService{}
+			}
+
+			librarypanels.NewService = func(store *sqlstore.SQLStore) librarypanels.LibraryPanelService {
+				return &mockLibraryPanelService{}
+			}
```

Comment:
Stubs should be reset after test is over:
```suggestion
			origNewElementsService := libraryelements.NewService
			origNewPanelsService := librarypanels.NewService
			t.Cleanup(func() {
				libraryelements.NewService = origNewElementsService
				librarypanels.NewService = origNewPanelsService
			})

			libraryelements.NewService = func(store *sqlstore.SQLStore) libraryelements.LibraryElementService {
				return &mockLibraryElementService{}
			}

			librarypanels.NewService = func(store *sqlstore.SQLStore) librarypanels.LibraryPanelService {
				return &mockLibraryPanelService{}
			}
```

Reply:
Good catch will fix this!

Reply:
Fixed with a different approach with HttpServer instead.